src/libUtils/DnsUtils.cpp

```diff
@@ -0,0 +1,329 @@
+      continue;
+    }
+    break;  // If we are here, means success
+  }
+
+  if (isShuttingDown) return false;
+
+  if (errcode != 0) {
+    LOG_GENERAL(WARNING, "Unable to connect " << url << ", Err: " << errcode);
+    return false;
+  }
+
+  res = result;
+
+  auto &ipStrList = dnsCacheListDataMap[listType].ipList;
+  ipStrList.clear();
+
+  while (res) {
+    switch (res->ai_family) {
+      case AF_INET:
+        ptr = &((struct sockaddr_in *)res->ai_addr)->sin_addr;
+        break;
+      case AF_INET6:
+        ptr = &((struct sockaddr_in6 *)res->ai_addr)->sin6_addr;
+        break;
+    }
+
+    if (ptr) {
+      inet_ntop(res->ai_family, ptr, addrstr, 100);
+      LOG_GENERAL(DEBUG, "Address: " << addrstr);
+      ipStrList.emplace_back(addrstr);
+    }
+    res = res->ai_next;
+    ptr = nullptr;
+  }
+
+  freeaddrinfo(result);
+
+  return true;
+}
+
+// Pub key are stored in the TXT record
+bool QueryPubKeyFromUrl(bytes &output, const string &pubKeyUrl) {
+  LOG_MARKER();
+
+  unsigned char query_buffer[256];
+  int resLen = 0;
+
+  for (unsigned int retry = 0; retry < QUERY_DNS_MAX_TRIES; ++retry) {
+    if (isShuttingDown) return false;
+
+    resLen = res_query(pubKeyUrl.c_str(), C_IN, ns_t_txt, query_buffer,
+                       sizeof(query_buffer));
+    if (resLen < 0) {
+      LOG_GENERAL(WARNING, "Failed to query pub key from "
+                               << pubKeyUrl << " Retry: " << retry);
+      continue;
+    }
+    break;  // If we are here, means success
+  }
+
+  if (isShuttingDown) return false;
+
+  if (resLen < 0) {
+    LOG_GENERAL(WARNING, "Unable to obtain pubkey from " << pubKeyUrl
+                                                         << " Res: " << resLen);
+    return false;
+  }
+
+  ns_msg nsMsg;
+  ns_initparse(query_buffer, resLen, &nsMsg);
+
+  if (ns_msg_count(nsMsg, ns_s_an) <= 0) {
+    LOG_GENERAL(WARNING, "No data found from pub key from " << pubKeyUrl);
+    return false;
+  }
+
+  ns_rr rr;
+  ns_parserr(&nsMsg, ns_s_an, 0, &rr);
+  auto rrData = ns_rr_rdata(rr) + 1;
+
+  string p(rrData, rrData + rr.rdlength - 1);
+
+  if (p.compare("0") == 0 || p.empty()) {
+    LOG_GENERAL(WARNING, "Pubkey from dns is 0 or empty: " << pubKeyUrl);
+    return false;
+  }
+
+  if (!DataConversion::HexStrToUint8Vec(p, output)) {
+    LOG_GENERAL(WARNING, "Invalid data obtained from pub key " << pubKeyUrl);
+    return false;
+  }
+  return true;
+}
+
+void QueryDnsList(DnsListType listType) {
+  LOG_MARKER();
+  if (isShuttingDown) return;
+
+  auto url = dnsAddresses[listType];
+  if (url.empty()) {
+    LOG_GENERAL(INFO, "DNS address is empty for type " << (int)listType);
+    return;
+  }
+
+  auto &dataListCache = dnsCacheListDataMap[listType];
+
+  if (!dataListCache.dataAccessMutex.try_lock()) {
+    LOG_GENERAL(INFO, "Another thread is querying " << dnsAddresses[listType]);
+    return;
+  }
+
+  if (!QueryIpStrListFromDns(listType)) {
+    LOG_GENERAL(WARNING, "Failed to obtain ip list from "
```

Comment:
Minor request, please make consistent the logs around "IP" and "DNS" (e.g., IP vs ip, Dns vs DNS), because some editors we use during debugging logs are case-sensitive.

(My personal preference is "IP" and "DNS").

(Also, if you agree with that, then maybe the files and function and variable names can also be modified for consistency, e.g., DNSUtils.cpp, currentIPKeys, etc.)